solidity/contracts/Coordinator.sol

```diff
@@ -28,16 +27,45 @@ contract Coordinator {
     uint256 _payment,
     uint256 _expiration,
     address[] _oracles,
+    uint8[] _vs,
+    bytes32[] _rs,
+    bytes32[] _ss,
     bytes32 _requestDigest
   ) public
   {
-    bytes32 id = getId(_payment, _expiration, _oracles, _requestDigest);
+    require(
+      (_oracles.length == _vs.length) && (_vs.length == _rs.length) && (_rs.length == _ss.length),
+      "Must pass in as many signatures as oracles");
+
+    bytes32 _serviceAgreementID = getId(_payment, _expiration, _oracles, _requestDigest);
```

Comment:
This should be `serviceAgreementID` as the convention is to use underscores for function parameters